src/AppInstallerRepositoryCore/Rest/Schema/Json/SearchRequestSerializer.cpp

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "Rest/Schema/IRestClient.h"
+#include "SearchRequestSerializer.h"
+#include <cpprest/json.h>
+#include "JsonHelper.h"
+#include "CommonRestConstants.h"
+
+namespace AppInstaller::Repository::Rest::Schema::Json
+{
+    namespace
+    {
+        // Search request constants
+        constexpr std::string_view Query = "Query"sv;
+        constexpr std::string_view Filters = "Filters"sv;
+        constexpr std::string_view Inclusions = "Inclusions"sv;
+        constexpr std::string_view MaximumResults = "MaximumResults"sv;
+        constexpr std::string_view RequestMatch = "RequestMatch"sv;
+        constexpr std::string_view KeyWord = "KeyWord"sv;
+        constexpr std::string_view MatchType = "MatchType"sv;
+        constexpr std::string_view PackageMatchField = "PackageMatchField"sv;
+        constexpr std::string_view FetchAllManifests = "FetchAllManifests"sv;
+    }
+
+    web::json::value SearchRequestSerializer::Serialize(const SearchRequest& searchRequest) const
+    {
+        std::optional<web::json::value> result = SerializeSearchRequest(searchRequest);
+
+        THROW_HR_IF(APPINSTALLER_CLI_ERROR_RESTSOURCE_INTERNAL_ERROR, !result);
+
+        return result.value();
+    }
+
+    std::optional<web::json::value> SearchRequestSerializer::SerializeSearchRequest(const SearchRequest& searchRequest) const
+    {
+        try
+        {
+            web::json::value json_body;
+            if (searchRequest.MaximumResults > 0)
+            {
+                json_body[JsonHelper::GetUtilityString(MaximumResults)] = searchRequest.MaximumResults;
+            }
+
+            if (searchRequest.IsForEverything())
+            {
+                json_body[JsonHelper::GetUtilityString(FetchAllManifests)] = web::json::value::string(L"true");
+                return json_body;
+            }
+
+            if (searchRequest.Query)
+            {
+                auto& requestMatch = searchRequest.Query.value();
+                web::json::value requestMatchObject = web::json::value::object();
+                requestMatchObject[JsonHelper::GetUtilityString(RequestMatch)] = GetRequestMatchJsonObject(requestMatch);
+                json_body[JsonHelper::GetUtilityString(Query)] = requestMatchObject;
+            }
+
+            if (searchRequest.Filters.size() > 0)

```

Comment:
>searchRequest.Filters.size() > 0 [](start = 16, length = 32)

While not a big deal for a vector, generally it is more efficient to use `!container.empty()` for a check like this.  Other containers don't necessarily store their size, and calculating it can be expensive compared to the empty check.